### PR TITLE
chore: add more @ts-check comments

### PR DESCRIPTION
**Pull request by NotWoods**
For #1905

Updates the JSDoc types to include more `// @ts-check` comments.

Includes some fixes that change behaviour:
- w3c/linter now properly imports the privsec rule.
- w3c/utils: IDBKeyVal now waits on `tx.done` instead of the invalid `tx.complete`.
- ui/about-respec wraps a `toFixed` call in `parseFloat`. This makes a type coercion from string to float more explicit.

### CODE REVIEW

File: src/w3c/linter.js [hunk 1]
```diff
@@ -6,6 +7,6 @@
 export const name = "w3c/linter";
 
 import linter from "../core/linter.js";
-import { lint as privsec } from "./linter-rules/privsec-section.js";
+import { rule as privsec } from "../core/linter-rules/privsec-section.js";
```
[the remark this answers is not in the document]
Comment (saschanaz) [reply src/w3c/linter.js]: @marcoscaceres Turns out we don't really use `w3c/linter`, should we remove this?

[the remark this answers is not in the document]
Comment (marcoscaceres) [reply src/w3c/linter.js]: Oops... no, we want this... It's supposed to be yelling at people about privacy and security sections.

[the remark this answers is not in the document]
Comment (marcoscaceres) [reply src/w3c/linter.js]: (we can handle this as a different bug if need be)

[the remark this answers is not in the document]
Comment (saschanaz) [reply src/w3c/linter.js]: Hmm, we should find which PR removed the dependency and blame the author then 🤣

[the remark this answers is not in the document]
Comment (marcoscaceres) [reply src/w3c/linter.js]: hahah... (whispers  "please let it not be me"🤞)... hahaha 😅

File: src/ui/search-specref.js [hunk 2]
```diff
@@ -63,6 +64,7 @@ function resultProcessor({ includeVersions } = { includeVersions: false }) {
       (collector, resultObj) => Object.assign(collector, resultObj),
       {}
     );
+    /** @type {Map<string, any>} */
```
[the remark this answers is not in the document]
Comment (saschanaz) [reply src/ui/search-specref.js]: https://github.com/microsoft/TypeScript/issues/33891

**Comment by saschanaz** [comment]
Force pushing forces me to review all 45 files again 😭 but let me try...

File: src/core/utils.js [hunk 3]
```diff
@@ -201,10 +201,11 @@ export class IDBKeyVal {
 
   /**
    * @param {string[]} keys
-   * @returns {[string, any][]}
+   * @returns {Promise<[string, any][]>}
    */
   async getMany(keys) {
     const keySet = new Set(keys);
+    /** @type {[string, any][]} */
```
Comment (saschanaz) [inline src/core/utils.js]: I think this should ultimately be `Map`, will open a separate PR.

**Pull request merged**